src/ring_theory/graded_algebra/basic.lean

```diff
@@ -77,4 +77,35 @@ def graded_algebra.decompose : A ≃ₐ[R] ⨁ i, 𝒜 i := alg_equiv.symm
   (graded_algebra.decompose 𝒜).symm (direct_sum.of _ i x) = x :=
 direct_sum.submodule_coe_alg_hom_of 𝒜 _ _
 
+/-- The projection maps of graded algebra-/
+def graded_algebra.proj (𝒜 : ι → submodule R A) [graded_algebra 𝒜] (i : ι) : A →+ A :=
+(𝒜 i).subtype.to_add_monoid_hom.comp $
+  (dfinsupp.eval_add_monoid_hom i).comp $
+  (graded_algebra.decompose 𝒜).to_alg_hom.to_ring_hom.to_add_monoid_hom
```

Comment:
This should be an `R`-linear map now that we're working with algebras

Reply:
```suggestion
def graded_algebra.proj (𝒜 : ι → submodule R A) [graded_algebra 𝒜] (i : ι) : A →ₗ[R] A :=
(𝒜 i).subtype.comp $
  (dfinsupp.lapply i).comp $
  (graded_algebra.decompose 𝒜).to_alg_hom.to_linear_map
```

Reply:
Thank you! I didn't know about `dfinsupp.lapply` and have to write
```lean
{ to_fun := λ a, ((graded_algebra.decompose 𝒜 a) i).1,
  map_add' := λ a a', by simp only [alg_equiv.map_add, submodule.coe_add,
    direct_sum.add_apply, subtype.val_eq_coe],
  map_smul' := λ r a, begin
    rw ring_hom.id_apply,
    erw alg_hom.map_smul (graded_algebra.decompose 𝒜).to_alg_hom r,
    simp only [dfinsupp.coe_smul, alg_equiv.coe_alg_hom, alg_equiv.to_alg_hom_eq_coe,
      submodule.coe_smul_of_tower, pi.smul_apply, subtype.val_eq_coe]
  end }
```

This is much cleaner